qiskit_dynamics/signals/__init__.py

```diff
@@ -12,14 +12,195 @@
+while :class:`~qiskit_dynamics.signals.DiscreteSignal` is internally implemented
+with array-based operations and geared more towards performance.
+
+
+Signal
+^^^^^^
+
+A :class:`~qiskit_dynamics.signals.Signal` is instantiated with an envelope.
+
+.. code-block:: python
+
+    f = lambda t: t**2
+    signal = Signal(envelope=f, carrier_freq=2.)
+
+**Important:** :class:`~qiskit_dynamics.signals.Signal` assumes the envelope ``f`` is
+*array vectorized* in the sense that ``f`` can operate on arrays of arbitrary shape
+and satisfy ``f(x)[idx] == f(x[idx])`` for a multidimensional index ``idx``. This
+can be ensured either by writing ``f`` to be vectorized, or by using the ``vectorize``
+function in ``numpy`` or ``jax.numpy``, e.g.:
+
+.. code-block:: python
+
+    import numpy as np
+    vectorized_f = np.vectorize(f)
+    signal = Signal(envelope=vectorized_f, carrier_freq=2.)
+
+Once instantiated, various functions associated with the siganl can be evaluated:
+
+.. code-block:: python
+
+    t = 0.5
+    # evaluate the full expression
+    signal(t)
+    # evaluate the envelope
+    signal.envelope(t)
+    # evaluate the complex value
+    signal.complex_value(t)
+
+We can also draw any of these functions over an interval:
+
+.. jupyter-execute::
+
+    from qiskit_dynamics.signals import Signal
+    signal = Signal(envelope=lambda t: t**2, carrier_freq=2.)
+
+    # draw with start point endpoint, and number of samples
+    signal.draw(t0=0., tf=2., n=100)
+
+
+Constant Signal
+^^^^^^^^^^^^^^^
+
+:class:`~qiskit_dynamics.signals.Signal` supports specification of a *constant signal*:
+
+.. code-block:: python
+
+    const = Signal(2.)
+
+This initializes the object to always return the constant ``2.``, and allows constants to be
+treated on the same footing as arbitrary :class:`~qiskit_dynamics.signals.Signal` instances.
+A :class:`~qiskit_dynamics.signals.Signal` operating in constant-mode can be checked via the
+boolean attribute ``const.is_constant``.
+
+
+DiscreteSignal
+^^^^^^^^^^^^^^
+
+A :class:`~qiskit_dynamics.signals.DiscreteSignal`  is specified
+as an array of samples with a sample width and an optional start point (defaults to ``0.``).
+:class:`~qiskit_dynamics.signals.DiscreteSignal` inherits the same API as
+:class:`~qiskit_dynamics.signals.Signal`, and once instantiated can be interacted with
+in the same way.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics.signals import DiscreteSignal
+    from matplotlib import pyplot as plt
+
+    discrete_signal = DiscreteSignal(dt=0.5, samples=[1., 2., 3.], carrier_freq=1.)
+
+    fig, axs = plt.subplots(1, 2, figsize=(14, 4))
+    discrete_signal.draw(t0=0., tf=1.5, n=100,
+                         axis=axs[0], title='Signal')
+    discrete_signal.draw(t0=0., tf=1.5, n=100,
+                         function='complex_value', axis=axs[1], title='Complex value')
+
+Note that the carrier wave is still analog for :class:`~qiskit_dynamics.signals.DiscreteSignal`.
```

Comment:
What does this mean?

Reply:
I've removed this as I agree that it's confusing, and what I meant to say is already implied by the general definition.